src/Resources/contao/controllers/FrontendIndex.php

```diff
@@ -260,17 +261,22 @@ public function renderPage($pageModel)
 			}
 		}
 
-		// Authenticate the user
-		if ($objPage->protected && !\System::getContainer()->get('security.authorization_checker')->isGranted('ROLE_MEMBER'))
-		{
-			throw new AccessDeniedException('Access denied: ' . \Environment::get('uri'));
-		}
-
-		// Check the user groups if the page is protected
+		// Authenticate the user if the page is protected
 		if ($objPage->protected)
 		{
+			if (!\System::getContainer()->get('contao.security.token_checker')->hasFrontendUser())
```

Comment:
This service should not be used in this case. It's only meant to check users across firewalls (because it unserialized the token from the session instead of using the token storage).